docker/dolfinx-complex-mode

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
-export LD_LIBRARY_PATH=/usr/local/dolfinx-complex/lib:$LD_LIBRARY_PATH
 export PATH=/usr/local/dolfinx-complex/bin:$PATH
 export PKG_CONFIG_PATH=/usr/local/dolfinx-complex/lib/pkgconfig:$PKG_CONFIG_PATH
 export PETSC_ARCH=linux-gnu-complex-32
```

Comment:
I would expect a corresponding `PETSC_DIR` to be consistent with how the PETSc install system is intended to be used.

Reply:
I managed to fix this before but forgot to remove the hacky changes to LD_LIBRARY_PATH, will remove and test.